addons/crm_survey/tests/test_crm_survey.py

```diff
@@ -0,0 +1,257 @@
+        login = "survey_manager"
+        # Adding in a sales team
+        if is_in_sales_team:
+            sales_team = self.env['crm.team'].create({
+                'name': 'Odoo Survey Team',
+                'use_leads': True
+            })
+            user = self.env['res.users'].search([('login', '=', login)])
+            sales_team.member_ids = [(4, user.id)]
+
+        with self.with_user(login):
+            # Create lead qualification survey
+            if survey_name is None:
+                survey_name = 'Questionnaire for the position of software developer'
+            survey = self.env['survey.survey'].create({
+                'title': survey_name,
+                'survey_type': survey_type,
+                'questions_layout': 'page_per_question',
+                'access_mode': 'public',
+                'users_login_required': False,
+            })
+
+            # Create questions
+            q01 = self._add_question(
+                None, 'How old are you?', 'simple_choice',
+                sequence=1,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                labels=[
+                    {'value': '18-30', 'is_create_lead': False},
+                    {'value': '30-50', 'is_create_lead': False},
+                    {'value': '50+', 'is_create_lead': False},
+                ])
+
+            q02 = self._add_question(
+                None, 'What programming languages do you use on a daily basis?', 'multiple_choice',
+                sequence=2,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                labels=[
+                    {'value': 'Assembly', 'is_create_lead': True},
+                    {'value': 'Java', 'is_create_lead': False},
+                    {'value': 'C', 'is_create_lead': False},
+                    {'value': 'Python', 'is_create_lead': False},
+                ])
+
+            q03 = self._add_question(
+                None, 'How many years of experience do you have in this position?', 'simple_choice',
+                sequence=3,
+                constr_mandatory=True, constr_error_msg='Please select an answer', survey_id=survey.id,
+                labels=[
+                    {'value': '0-1 year', 'is_create_lead': True},  # Newbie power
+                    {'value': '2-5 years', 'is_create_lead': False},
+                    {'value': '6+ years', 'is_create_lead': False},
+                ])
+
+            q04 = self._add_question(
+                None, 'Please skip this one', 'simple_choice',
+                sequence=4,
+                constr_mandatory=False, survey_id=survey.id,
+                labels=[
+                    {'value': 'No.', 'is_create_lead': False},
+                    {'value': 'NO', 'is_create_lead': False},
+                    {'value': 'OK, btw', 'is_create_lead': False},
+                ])
```

Comment:
We can probably remove one of the two, right?

Reply:
Actually I would keep only one of the questions with 'simple_choice' type. If matrix has a custom behavior, maybe we can add one instead?